holoviews/element/comparison.py

```diff
@@ -648,3 +656,140 @@ def __init__(self, *args, **kwargs):
+    """
+    ref_files = sorted(ref_reader.list_members())
+
+    matches = OrderedDict()
+    for ref_file in ref_files:
+        matches[ref_file] = ref_file if test_reader.member_exists(ref_file) else None
+
+    extra = set(test_reader.list_members()) - set(ref_files)
+    return matches, extra
+
+
+class ArchiveComparison(param.Parameterized):
+    """
+    Compare a 'test' archive against a 'reference' archive.
+
+    Use equal() to find out if the archives match or not,
+    or step through comparisons() for information about
+    each comparison performed.
+
+    Specify a custom match_fn to control exactly what (and 
+    in what order) to compare.
+
+    By default, archive members will simply be opened and read for
+    comparison; specify alternatives for particular file types
+    by adding them to load_fns.
+    """
+
+    match_fn = param.Callable(default=match_by_name,doc="""
+        Function that matches up files in a 'reference' archive to
+        those in a 'test' archive.
+
+        Should accept two archive readers and return:
+
+            1. a dict-like mapping of files to compare (using a value 
+               of None to indicate missing test data);
+
+            2. a list (potentially empty) of files present in the test
+               that are not present in the reference.
+        """)
+
+    load_fns = param.Dict(default=defaultdict(lambda: read, {'.hvz': loadpickle}))
+
+    def __init__(self,ref,test,**params):
+        super(ArchiveComparison,self).__init__(**params)
+        self.ref = self._read_archive(ref)
+        self.test = self._read_archive(test)        
+
+
+    def equal(self):
+        """
+        Return True if the two archives contain the same contents,
+        otherwise False.
+        
+        Stops at first failure to match (subsequently available as
+        .first_failure).
+        """
+        for result in self.comparisons():
+            if not result.match:
+                self.first_failure = result
+                return False
+        return True
+
+
+    def comparisons(self):
+        """
+        Return a generator to allow stepping through comparisons.
+
+        Each comparison yields a ComparisonResult.
+        """
+        matches,extra = self.match_fn(self.ref,self.test)
+
+        for unused in extra:
+            yield self._result(unused,False,'Extra test member')
+        
+        for rmember in matches:
+            if matches[rmember] is None:
+                yield self._result(rmember,False,'Test is missing member')
+            else:
+                load_fn = self.load_fns[os.path.splitext(rmember)[1].lower()]
+                r = load_fn(self.ref.open_member(rmember))
+                try:
+                    t = load_fn(self.test.open_member(matches[rmember]))
+                except:
+                    yield self._result(rmember,False,'Cannot open test and ref members with same function (%s): %s'%(lf,traceback.format_exc()))
```

Comment:
Sounds like we want instances of some sort of parameterized function/class in order to avoid this sort of clash.